### PR TITLE
Program visibility to Selected Trusted Intermediaries

### PR DESCRIPTION
**Pull request by swatkat1**
### Description
Program visibility to Selected Trusted Intermediaries.

## Release notes

Program visibility to Selected Trusted Intermediaries. Adding the ability for a CiviForm Admin to select which TI Orgs will have visibility to programs.

#### General

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

#### User visible changes

- [x] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [ ] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [ ] Manually tested at 200% size
- [ ] Manually evaluated tab order

#### New Features

- [ ] Add new FeatureFlag env vars to `server/conf/helper/feature-flags.conf`
- [ ] Conditioned new functionality on a [FeatureFlag](https://docs.civiform.us/contributor-guide/developer-guide/feature-flags)
- [ ] Wrote browser tests with the feature flag off and on, etc.

### Issue(s) this completes

Fixes #4247

### CODE REVIEW

**Review by gwendolyngoetz — APPROVED** [review]

File: server/app/views/admin/programs/ProgramFormBuilder.java [hunk 1]
```diff
@@ -198,6 +223,55 @@ private FormTag buildProgramForm(
...
+                true ? BaseStyles.BORDER_SEATTLE_BLUE : "")
+            .with(
+                input()
+                    .withId(id)
+                    .withType("checkbox")
+                    .withName("tiGroups" + Path.ARRAY_SUFFIX)
+                    .withValue(String.valueOf(tiId))
+                    .withCondChecked(selected)
+                    //  .condAttr("aria-invalid", "true")
+                    // .condAttr( "aria-required", "true")
```
Comment (bion) [inline server/app/views/admin/programs/ProgramFormBuilder.java:265]: **nitpick:** please remove commented code

File: server/app/auth/ProgramAcls.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      if (currentAccount.getManagedByGroup().isPresent()) {
+        return tiProgramViewAcls.contains(currentAccount.getManagedByGroup().get().id);
+      }
+      return false;
+    }
+    return false;
+  }
+  //  public boolean hasProgramAcls() {
+  //    return tiProgramViewAcls.size()>0;
+  //  }
```
Comment (bion) [inline server/app/auth/ProgramAcls.java:55]: **nitpick:** please remove commented code

File: server/app/services/program/ProgramService.java [hunk 3]
```diff
@@ -111,7 +114,8 @@ ErrorAnd<ProgramDefinition, CiviFormError> createProgramDefinition(
       String externalLink,
       String displayMode,
       ProgramType programType,
-      Boolean isIntakeFormFeatureEnabled);
+      Boolean isIntakeFormFeatureEnabled,
+      List<Long> tiGroups);
```
Comment (bion) [inline server/app/services/program/ProgramService.java:118]: **nitpick:** here and L140, please use an `ImmutableList` https://docs.civiform.us/contributor-guide/developer-guide/development-standards#prefer-using-immutable-collection-types

File: server/app/views/admin/programs/ProgramFormBuilder.java [hunk 4]
```diff
@@ -198,6 +223,55 @@ private FormTag buildProgramForm(
     return formTag;
   }
 
+  private DomContent showTISelectionList(List<Long> selectedTi, boolean selectTIChecked) {
```
Comment (bion) [inline server/app/views/admin/programs/ProgramFormBuilder.java:226]: **nitpick:** https://google.github.io/styleguide/javaguide.html#s5.3-camel-case
```suggestion
  private DomContent showTiSelectionList(List<Long> selectedTi, boolean selectTIChecked) {
```

File: server/app/services/program/ProgramService.java (hunk 3 above)
Comment (swatkat1) [reply server/app/services/program/ProgramService.java:118]: @bion - Sure. Just a clarification- Should the change from List to immutable list be done in the controller?

Comment (bion) [reply server/app/services/program/ProgramService.java:118]: It should be done closest to where the list originates

File: server/app/views/admin/programs/ProgramFormBuilder.java (hunk 1 above)
Comment (swatkat1) [reply server/app/views/admin/programs/ProgramFormBuilder.java]: Done!

File: server/app/auth/ProgramAcls.java (hunk 2 above)
Comment (swatkat1) [reply server/app/auth/ProgramAcls.java]: Done

File: server/app/services/program/ProgramService.java (hunk 3 above)
Comment (swatkat1) [reply server/app/services/program/ProgramService.java]: Done!

File: server/app/views/admin/programs/ProgramFormBuilder.java (hunk 4 above)
Comment (swatkat1) [reply server/app/views/admin/programs/ProgramFormBuilder.java]: Done!

**Pull request merged**